test/sqllogictest/ldbc_bi.slt

```diff
@@ -0,0 +1,4022 @@
+            ArrangeBy keys=[[#0]]
+              Get materialize.public.country
+            ArrangeBy keys=[[#0]]
+              Project (#1, #2)
+                Filter (#0 <= 2013-01-10 00:00:00 UTC) AND (2012-09-28 00:00:00 UTC <= #0)
+                  Get materialize.public.person_knows_person
+
+Used Indexes:
+  - materialize.public.person_id
+  - materialize.public.person_knows_person_person1id_person2id
+  - materialize.public.country_id
+  - materialize.public.city_id
+
+EOF
+
+######################################################################
+# QUERY 12
+######################################################################
+
+# \set startDate '\'2012-06-03\''::timestamp
+# \set lengthThreshold '120'
+# \set languages '\'{es, ta, pt}\''::varchar[]
+
+query T multiline
+EXPLAIN WITH
+  matching_message AS (
+    SELECT MessageId,
+           CreatorPersonId
+    FROM Message
+    WHERE Message.content IS NOT NULL
+      AND Message.length < 120
+      AND Message.creationDate > '2012-06-03'::TIMESTAMP
+      AND Message.RootPostLanguage = ANY ('{es, ta, pt}'::varchar[]) -- MZ change to use postgres containment check
+  ),
+  person_w_posts AS (
+    SELECT Person.id, count(matching_message.MessageId) as messageCount
+      FROM Person
+      LEFT JOIN matching_message
+        ON Person.id = matching_message.CreatorPersonId
+     GROUP BY Person.id
+  ),
+  message_count_distribution AS (
+    SELECT pp.messageCount, count(*) as personCount
+      FROM person_w_posts pp
+     GROUP BY pp.messageCount
+     ORDER BY personCount DESC, messageCount DESC
+  )
+SELECT *
+  FROM message_count_distribution
+ORDER BY personCount DESC, messageCount DESC
+----
+Explained Query:
+  Finish order_by=[#1 desc nulls_first, #0 desc nulls_first] output=[#0, #1]
+    Return
+      Reduce group_by=[#0] aggregates=[count(*)]
+        Project (#1)
+          Reduce group_by=[#0] aggregates=[count(#1)]
+            Union
+              Map (null)
+                Union
+                  Negate
+                    Project (#1)
+                      Join on=(#1 = #11) type=differential
+                        Get l0
+                        ArrangeBy keys=[[#0]]
+                          Distinct group_by=[#0]
+                            Project (#0)
+                              Get l1
+                  Project (#1)
+                    Get materialize.public.person
+              Get l1
+    With
+      cte l1 =
+        Project (#1, #12)
+          Filter (#19 < 120) AND (#11 > 2012-06-03 00:00:00 UTC) AND (#1) IS NOT NULL AND (#15) IS NOT NULL
+            Join on=(#1 = #20 AND #14 = #24) type=differential
+              Get l0
+              ArrangeBy keys=[[#9]]
+                Get materialize.public.message
+              ArrangeBy keys=[[#0]]
+                Distinct group_by=[#0]
+                  Project (#0)
+                    Filter (#0 = varchar_to_text(#1))
+                      FlatMap unnest_array({"es", "ta", "pt"})
```

Comment:
Ouch. This is a really inefficient execution of `Message.RootPostLanguage = ANY ('{es, ta, pt}'::varchar[])`. I think even the Join with `message` again on line 1563 is still part of doing this.

A manual rewrite would be to do `Message.RootPostLanguage IN ('es', 'ta', 'pt')` in the SQL (if I'm reading this correctly), but it would be great if we did this rewrite automatically.

(And then we could also take advantage of an index on `Message.RootPostLanguage`.)

Reply:
This is a transliteration of the Umbra query, and we could probably do that rewrite pretty cheaply. But maybe we should also just have this optimization?

Reply:
It's tricky, because the pattern gets complicated by the time we are in MIR. But in HIR, the pattern is relatively simple. Maybe we could have something like `MirScalarExpr::reduce`, but for HirScalarExpr.